tests/Psl/Regex/EveryMatchTest.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Tests\Regex;
+
+use PHPUnit\Framework\TestCase;
+use Psl\Regex;
+use Psl\Type\TypeInterface;
+
+use function Psl\Regex\capture_groups;
+
+final class EveryMatchTest extends TestCase
+{
+    /**
+     * @dataProvider provideMatchingData
+     */
+    public function testMatching(
+        array $expected,
+        string $subject,
+        string $pattern,
+        TypeInterface $shape = null,
+        int $offset = 0
+    ): void {
+        static::assertSame($expected, Regex\every_match($subject, $pattern, $shape, $offset));
+    }
+
+    /**
+     * @dataProvider provideNonMatchingData
+     */
+    public function testNotMatching(string $subject, string $pattern, int $offset = 0)
+    {
+        static::assertNull(Regex\every_match($subject, $pattern, null, $offset));
+    }
+
+    public function testMatchingWithInvalidPattern(): void
+    {
+        $this->expectException(Regex\Exception\InvalidPatternException::class);
+        $this->expectExceptionMessage("No ending delimiter '/' found");
+
+        Regex\every_match('hello', '/hello');
+    }
+
+    public function testInvalidCaptureGroup(): void
+    {
+        $this->expectException(Regex\Exception\RuntimeException::class);
+        $this->expectExceptionMessage('Invalid capture groups');
+
+        Regex\every_match('hello', '/(hello)/', capture_groups(['doesnotexist']));
+    }
+
+    public function provideMatchingData(): iterable
+    {
+        yield [
+            [
+                [
+                    0 => 'PHP',
+                    1 => 'PHP',
+                ]
+            ],
+            'PHP is the web scripting language of choice.',
+            '/(php)/i',
+            capture_groups([0, 1])
+        ];
+        yield [
+            [
+                [
+                    0 => 'Hello world',
+                    1 => 'Hello',
+                ]
+            ],
+            'Hello world is the web scripting language of choice.',
+            '/(hello) world/i',
+            capture_groups([0, 1])
+        ];
+        yield [
+            [
+                [
+                    0 => 'web',
+                    1 => 'web',
+                ]
+            ],
+            'PHP is the web scripting language of choice.',
+            '/(\bweb\b)/i',
+            capture_groups([0, 1])
+        ];
+        yield [
+            [
+                [
+                    0 => 'web',
+                    1 => 'web',
+                ]
+            ],
+            'PHP is the web scripting language of choice.',
+            '/(\bweb\b)/i'
+        ];
+        yield [
+            [
+                [
+                    0 => 'PHP',
+                    'language' => 'PHP'
+                ],
+            ],
+            'PHP is the web scripting language of choice.',
+            '/(?P<language>PHP)/',
+            capture_groups([0, 'language'])
```

Comment:
`0` should be added by capture_groups, right?
```suggestion
            capture_groups(['language'])
```

Reply:
Yes, it is added by default.
Doubles are filtered out, so it doesnt really matter if you add it or not - besides being more explicit.